community/kernel/src/main/java/org/neo4j/kernel/impl/store/kvstore/AbstractKeyValueStore.java

```diff
@@ -84,7 +84,21 @@ public String toString()
     protected final <Value> Value lookup( Key key, Reader<Value> reader ) throws IOException
     {
         ValueLookup<Value> lookup = new ValueLookup<>( reader );
-        return lookup.value( !state.lookup( key, lookup ) );
+        while ( true )
+        {
+            ProgressiveState<Key> originalState = this.state;
+            try
+            {
+                return lookup.value( !this.state.lookup( key, lookup ) );
+            }
+            catch ( IllegalStateException e )
```

Comment:
Are we sure it is `IllegalStateException` that gets thrown in the case of the state being closed?